executor/adapter.go

```diff
@@ -900,8 +934,127 @@ var (
 	sessionExecuteRunDurationInternal = metrics.SessionExecuteRunDuration.WithLabelValues(metrics.LblInternal)
 	sessionExecuteRunDurationGeneral  = metrics.SessionExecuteRunDuration.WithLabelValues(metrics.LblGeneral)
 	totalTiFlashQuerySuccCounter      = metrics.TiFlashQueryTotalCounter.WithLabelValues("", metrics.LblOK)
+
+	execBuildLocking = metrics.ExecPhaseDuration.WithLabelValues("build:locking", "0")
+	execOpenLocking  = metrics.ExecPhaseDuration.WithLabelValues("open:locking", "0")
+	execNextLocking  = metrics.ExecPhaseDuration.WithLabelValues("next:locking", "0")
+	execLockLocking  = metrics.ExecPhaseDuration.WithLabelValues("lock:locking", "0")
+	execBuildFinal   = metrics.ExecPhaseDuration.WithLabelValues("build:final", "0")
+	execOpenFinal    = metrics.ExecPhaseDuration.WithLabelValues("open:final", "0")
+	execNextFinal    = metrics.ExecPhaseDuration.WithLabelValues("next:final", "0")
+	execLockFinal    = metrics.ExecPhaseDuration.WithLabelValues("lock:final", "0")
```

Comment:
For the `next` operation maybe we need to integrate more with the kv client metrics in the future.